src/utils/optionsMatches.js

```diff
@@ -0,0 +1,42 @@
+import matchesStringOrRegExp from "./matchesStringOrRegExp"
+
+const extractValueFromInput = (property, input) => {
```

Comment:
I'm in favor of leaving this function out, and simply passing the value into the function directly. Each rule will know exactly how to access the value that it needs to check, and may need some special case; whereas this function locks us into a finite set of properties, making it a little more difficult to add a new one, and will cause problems if/when some rule needs a special case.